### PR TITLE
set the lower bound based on the parent node if given

### PR DESCRIPTION
**Pull request by Wikunia**
Based on the issue #16 I propose the following changes @dhendryc Cc: @matbesancon 

When adding a new node the lower bound is set as the lower bound of the parent node if one is given (always just not for the root node). I've created a simple test to verify that this is true.
Maybe an actual test case for the issue written in #16 would be great as well.

Let me know what you think :)

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -7,6 +7,10 @@ using Cbc
...
     
 
-@testset "Bonobo.jl" begin
+@testset "End2End tests" begin
     include("end2end/mip.jl")
 end
+
+@testset "Unit tests" begin
+    include("unit/add_node.jl")
+end
```
Comment (matbesancon) [inline test/runtests.jl:16]:
```suggestion
end

```
newline

File: test/unit/add_node.jl [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    BB.set_root!(bnb_model, NamedTuple())
+    root = BB.get_next_node(bnb_model, BB.BFS())
+    lb, ub = BB.evaluate_node!(bnb_model, root)
+    BB.set_node_bound!(bnb_model.sense, root, lb, ub)
+
+    # adding a new node with the root as parent
+    BB.add_node!(bnb_model, root, NamedTuple())
+    last_node_id = bnb_model.num_nodes
+    @test bnb_model.nodes[last_node_id].lb == 0.0
+end
```
Comment (matbesancon) [inline test/unit/add_node.jl:29]:
```suggestion
end

```

**Comment by matbesancon** [comment]
LGTM!

**Comment by dhendryc** [comment]
@Wikunia thank you for fixing that! I only got around testing my example today and realized there is another issue related to this causing us to still evaluate more nodes than necessary. I did not think about that when proposing #16 : We should also not branch on nodes with greater lower bound. So we should avoid something like this:

The Solution at node 23 is [1.0, 1.0, 1.0, 1.0, 1.0] with primal 0.4000000000000001
The Solution at node 26 is [0.0, 1.0, 1.0, 0.0, 0.5999999994657994] with primal 0.52
Split index : 5 parent node : 26 children : 32 and 33 

I am testing this example: min f(x) = 0.5*( sum (x_i -0.6)² ) over the 0,1-hypercube where each variable is discrete. The expected solution is the all 1 vector. The prints are from within `evaluate_node!`  and `get_branching_nodes_info`. 
If we are only interested in one solution then there is no point in branching at node 26 because its lower bound is already worse than the tree incumbent. One could add an `if`-statement in `get_branching_variable` : 

`if node.lb >= tree.incumbent
  return -1`

Thank you and sorry for the late reply!

**Comment by Wikunia** [comment]
Thanks @dhendryc I'll look into it, hopefully tmr

**Pull request merged**